Mathlib/Analysis/Complex/Liouville.lean

```diff
@@ -79,6 +72,41 @@ theorem norm_deriv_le_of_forall_mem_sphere_norm_le {c : ℂ} {R C : ℝ} {f : 
       norm_deriv_le_aux hR (e.differentiable.comp_diffContOnCl hd) fun z hz =>
         (UniformSpace.Completion.norm_coe _).trans_le (hC z hz)
 
+/-- **Cauchy's estimate for derivatives**:  If `f` is complex differentiable on an open disc of
+radius `R > 0`, is continuous on its closure, and its values on the boundary circle of this disc
+are bounded from above by `C`, then the norm of its `n`-th derivative at the center is at most
+`n.factorial * C / R ^ n`. -/
+theorem norm_iteratedDeriv_le_of_forall_mem_sphere_norm_le [CompleteSpace F] {c : ℂ} {R C : ℝ}
+    {n : ℕ} {f : ℂ → F} (hR : 0 < R) (hf : DiffContOnCl ℂ f (ball c R))
+    (hC : ∀ z ∈ sphere c R, ‖f z‖ ≤ C) :
+    ‖iteratedDeriv n f c‖ ≤ n.factorial * C / R ^ n := by
+  have hp : ∀ z ∈ sphere c R, ‖(z - c)⁻¹ ^ (n + 1) • f z‖ ≤ C / (R ^ n  * R) :=
+    fun z (hz : ‖z - c‖ = R) => by
+    have := (div_le_div_iff_of_pos_right (mul_pos (pow_pos hR n) hR)).2 (hC z hz)
+    simp [norm_smul, norm_pow, norm_inv, hz, ← div_eq_inv_mul]
+    exact this
```

Comment:
You can introduce variables in a `have` statement, just like you can for theorems; then you don't need to intro them manually in the proof. Also, technically the `simp` here is non-terminal, which is against mathlib rules; but you can combine the `simp` and the following `exact` into a single statement with `simpa using`:
```suggestion
  have hp (z) (hz : ‖z - c‖ = R) : ‖(z - c)⁻¹ ^ (n + 1) • f z‖ ≤ C / (R ^ n  * R) := by
    simpa [norm_smul, hz, ← div_eq_inv_mul]
      using (div_le_div_iff_of_pos_right (mul_pos (pow_pos hR n) hR)).2 (hC z hz)
```